src/coreclr/jit/lsra.cpp

```diff
@@ -448,7 +448,11 @@ SingleTypeRegSet LinearScan::internalFloatRegCandidates()
     }
     else
     {
+#ifdef TARGET_AMD64
+        return RBM_FLT_CALLEE_TRASH.GetFloatRegSet();
+#else
         return RBM_FLT_CALLEE_TRASH;
+#endif // TARGET_AMD64
```

Comment:
Is the ifdef necessary?

Reply:
yeah, for amd64, `RBM_FLT_CALLEE_TRASH` maps to `get_RBM_FLT_CALLEE_TRASH()` which returns `regMaskTP` but for other platforms, it is just `uint64_t` or `SingleTypeRegSet`.

Reply:
I will see if I can unify this in future as this work evolves.